src/Rings/mpoly.jl

```diff
@@ -1294,3 +1294,20 @@ end
 
 # assure compatibility with generic code for MPolyQuos:
 lift(f::MPolyRingElem) = f
+
+### Hessian matrix
+function hessian_matrix(f::MPolyElem)
+  R = parent(f)
+  n = nvars(R)
+  df = jacobi_matrix(f)
+  result = zero(MatrixSpace(R, n, n))
```

Comment:
```suggestion
  result = zero_matrix(R, n, n)
```
This saves the creation of the matrix space.

Reply:
Thanks! I was not aware of this constructor. Probably this has to do with the rather recent decision to drop parent objects for matrices over rings? Either way, this is reasonable.

Reply:
This has been there for a few years

Reply:
And which "decision to drop parent objects for matrices over rings" ? I am not aware of such a decision.